src/psyclone/tests/psyir/transformations/parallel_loop_trans_test.py

```diff
@@ -103,9 +104,24 @@ def test_paralooptrans_validate_ignore_dependencies_for(fortran_reader,
     assert ("Loop cannot be parallelised because the dependency analysis "
             "reported:\nWarning: Variable 'sum' is read first, which indicates"
             " a reduction. Variable: 'sum'.") in str(err.value)
-    # With the verbose option, the dependency issue will be log as a comment
-    assert ("! PSyclone: Loop cannot be parallelised because the dependency"
-            " analysis reported:" in fortran_writer(psyir))
+    # With the verbose option, the dependency will be reported in a comment
+    assert ("PSyclone: Loop cannot be parallelised because the dependency"
+            " analysis reported:" in loop.preceding_comment)
+
+    # Test that the inner loop does not log again the same error message
+    with pytest.raises(TransformationError) as err:
+        trans.validate(loop.loop_body[0], options={"verbose": True})
+    assert ("PSyclone: Loop cannot be parallelised because the dependency"
+            not in loop.loop_body[0].preceding_comment)
+
+    # But if it's not already in an ancestor, it adds the comment there
+    loop.preceding_comment = ""
+    with pytest.raises(TransformationError) as err:
+        trans.validate(loop.loop_body[0], options={"verbose": True})
```

Comment:
This has made me realise that we have slightly non-intuitive behaviour here - calling `validate` on a Node can actually modify it (in the sense that `preceding_comment` can be modified) if `verbose` is set. I don't think there's any easy way of avoiding this but I thought I would flag it.

Reply:
This is the expected behaviour. Many verbose options now add comments during the validate. But I don't think this is a tree modification, just a logging mechanism using the preceding_comment mixing attribute.  I raised #2670 to formalise this pattern better and make it configurable: choose between comments/log/stderr.

Reply:
OK.